File: packages/cloud-adapters/secrets/aws.go
package secrets

import (
	"context"
	"errors"
)

// AwsAdapter is the AWS-Secrets-Manager-backed secrets adapter. Stubbed pending the AWS-cloud milestone.
type AwsAdapter struct{}

func NewAwsAdapter() *AwsAdapter { return &AwsAdapter{} }

func (*AwsAdapter) MintKey(context.Context, string) error {
	return errors.New("NotImplemented: cloud-adapters/secrets/aws.MintKey — Secrets Manager impl lands in the AWS-cloud milestone")
}
func (*AwsAdapter) Encrypt(context.Context, string, []byte) (string, error) {
	return "", errors.New("NotImplemented: cloud-adapters/secrets/aws.Encrypt — Secrets Manager impl lands in the AWS-cloud milestone")
}
func (*AwsAdapter) Decrypt(context.Context, string, string) ([]byte, error) {
	return nil, errors.New("NotImplemented: cloud-adapters/secrets/aws.Decrypt — Secrets Manager impl lands in the AWS-cloud milestone")
}
func (*AwsAdapter) RotateKey(context.Context, string) error {
	return errors.New("NotImplemented: cloud-adapters/secrets/aws.RotateKey — Secrets Manager impl lands in the AWS-cloud milestone")
}
func (*AwsAdapter) DeleteKey(context.Context, string) error {
	return errors.New("NotImplemented: cloud-adapters/secrets/aws.DeleteKey — Secrets Manager impl lands in the AWS-cloud milestone")
}

File: packages/cloud-adapters/secrets/aws.ts
import type { SecretsAdapter } from "./interface";

const notImpl = (op: string): never => {
  throw new Error(`NotImplemented: cloud-adapters/secrets/aws.${op} — Secrets Manager impl lands in the AWS-cloud milestone`);
};

export const aws: SecretsAdapter = {
  mintKey:   async () => notImpl("mintKey"),
  encrypt:   async () => notImpl("encrypt"),
  decrypt:   async () => notImpl("decrypt"),
  rotateKey: async () => notImpl("rotateKey"),
  deleteKey: async () => notImpl("deleteKey"),
};
